packages/fx-core/src/plugins/resource/appstudio/utils/telemetry.ts

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { PluginContext, SystemError, UserError } from "@microsoft/teamsfx-api";
+import { Constants } from "./../constants";
+
+export enum TelemetryPropertyKey {
+    component = "component",
+    errorType = "error-type",
+    errorCode = "error-code",
+    errorMessage = "error-message",
+    validationResult = "validation-result"
+}
+
+enum TelemetryPropertyValue {
+    UserError = "UserError",
+    SystemError = "SystemError",
+    UnhandledError = "UnhandledError"
+}
+
+export enum TelemetryEventName {
+    validateManifest = "validateManifest",
+    buildTeamsPackage = "buildTeamsPackage",
```

Comment:
snake case style. telemetry pipeline will make every thing lower case. And use "build-start" and "build" pair to address start and exit of operation.